src/Translation_unit.ml

```diff
@@ -122,25 +140,39 @@ let parse_print (XUnit xunit) (conf : Conf.t) ~input_name ~input_file ic
   (* iterate until formatting stabilizes *)
   let rec print_check ~i ~(conf : Conf.t) ~ast ~comments ~source_txt
       ~source_file : result =
+    let binary = Poly.(Conf.line_endings conf input_file = `Unix) in
     let tmp, oc =
       if not Conf.debug then Filename.open_temp_file ~temp_dir:dir base ext
       else
         let name = Format.sprintf "%s.%i%s" base i ext in
         let tmp = Filename.concat dir name in
         Format.eprintf "%s@\n%!" tmp ;
-        let oc = Out_channel.create ~fail_if_exists:(not Conf.debug) tmp in
+        let oc =
+          Out_channel.create ~binary ~fail_if_exists:(not Conf.debug) tmp
+        in
         (tmp, oc)
     in
     dump xunit dir base ".old" ".ast" ast ;
     let source = Source.create source_txt in
     let cmts_t = xunit.init_cmts source conf ast comments in
     let fs = Format.formatter_of_out_channel oc in
+    ( if
+      (Sys.unix && Poly.(Conf.line_endings conf input_file = `Windows))
+      || ((not Sys.unix) && crlf_translated_to_lf ())
```

Comment:
I don't understand the rational for this one

Reply:
The idea was to translate \n to \r\n when we are not on unix and the \r\n are not preserved when printing the formatted output (because we are not sure how they are handled on cygwin or windows).
I guess the `line_endings = windows` condition has been forgotten for this case.
Something like this should be better:
- on unix: if we want to produce windows output
- on windows/cygwin: if we want to produce windows output and \r\n are not preserved